iterator_methods.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    length.times do |i|
+      yield self[i]
+    end
+  end
+
+  def my_each_with_index
+    length.times do |i|
+      yield self[i], i
+    end
+  end
+
+  def my_select
+    aux_array = []
+    length.times do |i|
+      aux_array << self[i] if yield self[i]
+    end
+    aux_array
+  end
+
+  def my_all?
+    length.times do |i|
+      return false unless yield self[i]
+    end
+    true
+  end
+
+  def my_any?
+    length.times do |i|
+      return true if yield self[i]
+    end
+    false
+  end
+
+  def my_none?
+    length.times do |i|
+      return false if yield self[i]
+    end
+    true
+  end
+
+  def my_count
+    count = 0
+    length.times do |i|
+      count += 1 if yield self[i]
+    end
+    count
+  end
+
+  def my_map(*my_proc)
+    if my_proc.empty?
+      length.times do |i|
+        self[i] = yield self[i]
+      end
+    else
+      length.times do |i|
+        self[i] = my_proc[0].call(self[i])
+      end
+    end
+    self
+  end
+
+  def my_inject
+    memo = self[0]
+    (length - 1).times do |i|
+      memo = yield memo, self[i + 1]
+    end
+    memo
+  end
```

Comment:
Not working
```
> [5,6,7,8,9,10].inject(1) { |product, n| product * n }
 => 151200 
> [5,6,7,8,9,10].my_inject(1) { |product, n| product * n }
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):5
        1: from iterator_methods.rb:66:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Reply:
ok.. the method inject need arguments.. it's done